src/test/java/basesyntax/service/ReportGenerationImplTest.java

```diff
@@ -0,0 +1,43 @@
+package basesyntax.service;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.impl.ReportGenerationImpl;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ReportGenerationImplTest {
+    private static Map<Fruit, Integer> testMap;
+    private static StringBuilder testReport;
+    private static ReportGenerator testReportGenerator;
+
+    @BeforeClass
+    public static void setUp() {
+        testMap = new HashMap<>();
+        testMap.put(new Fruit("banana"), 152);
+        testMap.put(new Fruit("apple"), 90);
+        testReportGenerator = new ReportGenerationImpl();
+        testReport = new StringBuilder();
+        testReport.append("fruit,quantity")
+                .append(System.lineSeparator())
+                .append("banana,152")
+                .append(System.lineSeparator())
+                .append("apple,90")
+                .append(System.lineSeparator());
+    }
+
+    @Test
+    public void correctReportGeneration_ok() {
```

Comment:
Let's take more age case, try to generate report with empty storage, or with only one element